routing/index_graph_starter.cpp

```diff
@@ -109,37 +117,13 @@ set<NumMwmId> IndexGraphStarter::GetMwms() const
   return mwms;
 }
 
-bool IndexGraphStarter::DoesRouteCrossNonPassThrough(
-    RoutingResult<Segment, RouteWeight> const & result) const
+bool IndexGraphStarter::CheckWeight(RouteWeight const & weight) const
 {
-  int nonPassThroughCrossAllowed = 0;
-  auto isRealOrPart = [this](Segment const & segment) {
-    if (!IsFakeSegment(segment))
-      return true;
-    Segment dummy;
-    return m_fake.FindReal(segment, dummy);
-  };
-  auto isPassThroughAllowed = [this](Segment const & segment) {
-    auto real = segment;
-    bool const convertionResult = ConvertToReal(real);
-    CHECK(convertionResult, ());
-    return m_graph.IsPassThroughAllowed(real.GetMwmId(), real.GetFeatureId());
-  };
-
-  auto const firstRealOrPart = find_if(result.m_path.begin(), result.m_path.end(), isRealOrPart);
-  if (firstRealOrPart != result.m_path.end() && !isPassThroughAllowed(*firstRealOrPart))
-    ++nonPassThroughCrossAllowed;
-
-  auto const lastRealOrPart = find_if(result.m_path.rbegin(), result.m_path.rend(), isRealOrPart);
-  // If firstRealOrPart and lastRealOrPart point to the same segment increment
-  // nonPassThroughCrossAllowed once
-  if (lastRealOrPart != result.m_path.rend() && &*lastRealOrPart != &*firstRealOrPart &&
-      !isPassThroughAllowed(*lastRealOrPart))
-  {
-    ++nonPassThroughCrossAllowed;
-  }
+  int const nonPassThroughCrossAllowed =
+      (m_startPassThroughAllowed ? 0 : 1) + (m_finishPassThroughAllowed ? 0 : 1);
```

Comment:
Probably it's worth writing a comment here?

Reply:
done